src/test/java/org/breedinginsight/model/delta/DeltaEntityFactoryUnitTest.java

```diff
@@ -0,0 +1,461 @@
+
+        // Check that clone makes a correct copy
+        BrAPIObservationVariable clonedBrAPIObservationVariable = deltaObservationVariable.cloneBrAPIObject();
+        assertNotNull(clonedBrAPIObservationVariable);
+        assertEquals(clonedBrAPIObservationVariable, brAPIObservationVariable);
+    }
+
+    @Test
+    @SneakyThrows
+    void deltaEnvironmentTest() {
+        // Create BrAPIStudy
+        BrAPIStudy brAPIStudy = createBrAPIStudy();
+
+        // Use the factory to create a DeltaEnvironment from the BrAPIStudy
+        Environment environment = entityFactory.makeEnvironmentBean(brAPIStudy);
+
+        // Check that clone makes a correct copy
+        BrAPIStudy clonedBrAPIStudy = environment.cloneBrAPIObject();
+        assertNotNull(clonedBrAPIStudy);
+        assertEquals(clonedBrAPIStudy, brAPIStudy);
+    }
+
+    @Test
+    @SneakyThrows
+    void deltaExperimentTest() {
+        // Create BrAPITrial
+        BrAPITrial brAPITrial = createBrAPITrial();
+
+        // Use the factory to create a DeltaExperiment from the BrAPITrial
+        Experiment experiment = entityFactory.makeExperimentBean(brAPITrial);
+
+        // Check that clone makes a correct copy
+        BrAPITrial clonedBrAPITrial = experiment.cloneBrAPIObject();
+        assertNotNull(clonedBrAPITrial);
+        assertEquals(clonedBrAPITrial, brAPITrial);
+    }
+
+    private BrAPIGermplasm createBrAPIGermplasm() {
+        String additionalInfoString = "{\"additionalInfo\":{\"createdBy\":{\"userId\":\"101e7314-ba2c-466b-a1e0-f02409ab0d3d\",\"userName\":\"BI-DEV Admin\"},\"createdDate\":\"14/06/2024 19:17:40\",\"femaleParentUUID\":\"2927a4a5-c204-4255-850b-a1eb2c291263\",\"listEntryNumbers\":{\"fa0f1715-84b8-4ca7-8abc-ff191f221048\":\"38356\"},\"importEntryNumber\":\"38356\",\"maleParentUnknown\":false}}";
+
+        List<BrAPIGermplasmDonors> donors = new ArrayList<>();
+        donors.add(new BrAPIGermplasmDonors().donorAccessionNumber("abc").donorInstituteCode("institution"));
+        donors.add(new BrAPIGermplasmDonors().donorAccessionNumber("xyz").donorInstituteCode("institution"));
+
+        List<BrAPIGermplasmSynonyms> synonyms = new ArrayList<>();
+        synonyms.add(new BrAPIGermplasmSynonyms().synonym("germA").type("synonym"));
+        synonyms.add(new BrAPIGermplasmSynonyms().synonym("germB").type("synonym"));
+
+        List<BrAPITaxonID> taxonIds = new ArrayList<>();
+        taxonIds.add(new BrAPITaxonID().taxonId("testTaxon1").sourceName("test taxon source"));
+        taxonIds.add(new BrAPITaxonID().taxonId("testTaxon2").sourceName("test taxon source"));
+
+        var coordinates = BrApiGeoJSON.builder().geometry(null).type("Feature").build();
+        var origin = new BrAPIGermplasmOrigin().coordinates(coordinates).coordinateUncertainty("+/- 10");
+        var storageType = new BrAPIGermplasmStorageTypes(BrAPIGermplasmStorageTypesEnum._10);
```

Comment:
The cloning test fails for GermplasmEntity because the storage types are different.

Reply:
I implemented `equals` for `BrAPIGermplasmStorageTypes` in the [brapi PR that was merged](https://github.com/Breeding-Insight/brapi/pull/224/files), you may need to run `clean install -U` (see https://maven.apache.org/ref/3.9.8/maven-embedder/cli.html) or even purge your local maven cache (usually ~/.m2) if the new snapshot still isn't being picked up.